pkg/lib/kubernetes/pkg/util/hash/hash.go

```diff
@@ -17,21 +17,39 @@ limitations under the License.
 package hash
 
 import (
-	"hash"
-
-	"github.com/davecgh/go-spew/spew"
+	"crypto/sha256"
+	"encoding/json"
+	"fmt"
+	"math/big"
 )
 
 // DeepHashObject writes specified object to hash using the spew library
 // which follows pointers and prints actual values of the nested objects
 // ensuring the hash does not change when a pointer changes.
-func DeepHashObject(hasher hash.Hash, objectToWrite interface{}) {
+func DeepHashObject(obj interface{}) (string, error) {
+	// While the most accurate encoding we could do for Kubernetes objects (runtime.Object)
+	// would use the API machinery serializers, those operate over entire objects - and
+	// we often need to operate on snippets. Checking with the experts and the implementation,
+	// we can see that the serializers are a thin wrapper over json.Marshal for encoding:
+	// https://github.com/kubernetes/kubernetes/blob/8509ab82b96caa2365552efa08c8ba8baf11c5ec/staging/src/k8s.io/apimachinery/pkg/runtime/serializer/json/json.go#L216-L247
+	// Therefore, we can be confident that using json.Marshal() here will:
+	//  1. be stable & idempotent - the library sorts keys, etc.
+	//  2. be germane to our needs - only fields that serialize and are sent to the server
+	//     will be encoded
+
+	hasher := sha256.New224()
 	hasher.Reset()
```

Comment:
Nit: Not sure a `Reset()` is needed immediately after a `New224()`.

Reply:
`sha256.Sum224()` does it after allocating a new digest, so I figured I'd be good to follow their lead:

```go
// Sum224 returns the SHA224 checksum of the data.
func Sum224(data []byte) [Size224]byte {
	if boring.Enabled {
		return boring.SHA224(data)
	}
	var d digest
	d.is224 = true
	d.Reset()
	d.Write(data)
	sum := d.checkSum()
	ap := (*[Size224]byte)(sum[:])
	return *ap
}
```